server/bin/basket-proxy-server.js

```diff
@@ -0,0 +1,154 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var url = require('url');
+var mozlog = require('mozlog');
+var request = require('request');
+var p = require('bluebird');
+
+var config = require('../lib/configuration');
+mozlog.config(config.get('logging'));
+
+var logger = require('mozlog')('server.basketproxy');
+
+var express = require('express');
+var bodyParser = require('body-parser');
+var cors = require('cors');
+
+var CORS_ORIGIN = config.get('public_url');
+var API_KEY = config.get('basket.api_key');
+var API_URL = config.get('basket.api_url');
+var VERIFY_URL = config.get('oauth_url') + '/v1/verify';
+
+// Verify an OAuth token and return the associated credentials
+function verifyOAuthToken(cb) {
+  return function (req, res, next) {
+    var authHeader = req.headers && req.headers.authorization;
+
+    if (! authHeader) {
+      logger.error('unauthorized');
+      return next(new Error('Not authorized'));
+    }
+
+    var token = authHeader.replace(/^Bearer /, '');
+
+    request.post({
+      url: VERIFY_URL,
+      json: {
+        token: token
+      }
+    }, function (err, _, body) {
+      // TODO use an actual error format
+      if (err) {
+        logger.error('auth.error', err);
+        res.status(400).send(err);
+        return;
+      }
+
+      if (body.code >= 400) {
+        logger.error('unauthorized', body);
+        res.status(400).send('unauthorized');
+        return;
+      }
+
+      logger.debug('auth.valid', body);
+
+      cb(body, req, res, next);
+    });
+  };
+}
+
+// Send a request to the Basket backend
+function basketRequest(path, method, params) {
+  var config = {
+    url: API_URL + path,
+    strictSSL: true,
+    method: method,
+    headers: {
+      'X-API-Key': API_KEY
+    },
+    form: params
+  };
+
+  var deferred = p.defer();
+
+  request(config, function (err, httpRequest, body) {
+    if (err) {
+      logger.error('basket.error', err);
+      return deferred.reject(err);
+    }
+
+    if (httpRequest.statusCode !== 200) {
+      // TODO do something here.
+    }
+
+    deferred.resolve(body);
+  });
+
+  return deferred.promise;
+}
+
+function forwardBasketResponse(res, path, method, params) {
+  return basketRequest(path, method, params)
+    .then(function (body) {
+      res.type('application/json');
+      res.send(body);
+    }, function (err) {
+      res.status(400).send(err);
+    });
+}
+
+function initApp() {
+  var app = express();
+  app.use(bodyParser.json());
+
+  app.use(cors({
+    origin: CORS_ORIGIN
+  }));
+
+  app.get('/lookup-user', verifyOAuthToken(function (creds, req, res) {
+    forwardBasketResponse(res, '/lookup-user?email=' + creds.email, 'get');
+  }));
+
+  app.post('/subscribe', verifyOAuthToken(function (creds, req, res) {
+    forwardBasketResponse(res, '/subscribe/', 'post', {
+      email: creds.email,
+      newsletters: req.body.newsletters.join(','),
```

Comment:
`/subscribe/` also takes some other optional params like`country`, `lang`, and `source_url`. We'd take whatever you can send :) `source_url` is just so that we can see from where people are coming, but here we can set it to a constant value that will indicate that it came from FxA. Something like `https://accounts.firefox.com/` would be fine.